core/primitives/src/optimistic_block.rs

```diff
@@ -0,0 +1,76 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use near_crypto::Signature;
+
+use crate::block::BlockHeader;
+use crate::hash::{hash, CryptoHash};
+use crate::types::BlockHeight;
+
+#[derive(BorshSerialize, BorshDeserialize, Clone, Debug, Eq, PartialEq)]
+pub struct OptimisticBlock {
+    pub block_height: BlockHeight,
+    pub prev_block_hash: CryptoHash,
+    pub block_timestamp: u64,
+
+    // Data to confirm the correctness of randomness beacon output
+    pub random_value: CryptoHash,
+    pub vrf_value: near_crypto::vrf::Value,
+    pub vrf_proof: near_crypto::vrf::Proof,
+
+    /// Signature of the block producer.
+    pub signature: Signature,
+    pub hash: CryptoHash,
+}
+
+impl OptimisticBlock {
+    #[cfg(feature = "clock")]
+    pub fn new(
+        prev: &BlockHeader,
```

Comment:
Let's rename with `prev_block_header`, the namings for Block are not necessarily optimal.